bot/res/haskell

```diff
@@ -1,3574 +0,0 @@
-ToRunElems
-ToRunStyle
-ToRunStyleD
-ToSp
-ToStart
-ToStr
-ToStrAttr
-ToString
-ToStyle
-ToStyles
-ToTable
-ToTblGrid
-ToTblLook
-ToTexinfo
-ToTextile
-ToXml
-ToYaml
-TP
-T.pack
-Tr
-TR
-Trace
-TrackChanges
-TrackLine
-Trademark
-TrailingNewlines
-Transform
-Traversable
-Traversable.mapM
-Treat
-Tree
-T.replicate
-Trim
-Triple
-True
-Try
-TS
-T.singleton
-T.stripEnd
-Tt
-T.Text
-T.toUpper
-T.unpack
-TupleSections
-TWiki
-TWikiWithWarnings
-Twips
-Two
-TwoLines
-TwoParens
-TWParser
-Txt2Tags
-Txt2TagsNoMacros
-Type
-Type.
-Typeable
-TypeOf
-Types
-TypesDoc
-TypesEntry
-Typeset
-TypesList
-TypeStr
-TypeSynonymInstances
-TypeTable
-TZ
-Unable
-Uncategorized
-Undefined
-Underline
-Underscores
-Unescapes
-Unicode
-UnicodeChar
-Uniform
-Unindented
-Unique
-UniqueId
-UNIX
-Unknown
-UnknownTagType
-UnlessSmart
-Unqualified
-UnsignedByte
-UnsignedLong
-UnsignedRational
-UnsignedShort
-UntilHeaderAboveLevel
-UNTITLED
-Untyped
-UnwantedBlocks
-Up
-UPC
-Update
-Upper
-UpperAlpha
-Uppercase
-UpperRoman
-Uri
-URI
-URI.
-URIReference
-URIs
-URIs.
-URIString
-Url
-URL
-URL.
-URLs
-URLString
-URN
-US
-USA
-Use
-Used
-UsePrefix
-User
-UserComment
-UserDataDir
-USERNAME
-Uses
-UsesEuro
-UseTags
-UTCTime
-UTF
-Utf8
-UTF8
-UTF8.fromString
-UTF8.fromStringLazy
-UTF8.hPutStrLn
-UTF8.putStrLn
-UTF8.readFile
-UTF8.toString
-UTF8.toStringLazy
-UTF8.writeFile
-Utilities
-Utility
-UUID
-Val
-ValAttr
-ValidChar
-ValLowercase
-Value
-Values
-ValueToInlines
-ValueToJSON
-ValueToString
-Var
-Variable
-Variables
-Variant
-Varieties
-VarListEntry
-VBlock
-VBScript
-Verb
-VERB
-Verbatim
-VerbatimBlock
-VerbatimChar
-VerbInNote
-Verbose
-Verilog
-Version
-VertAlign
-Vertical
-VHDL
-ViewL
-ViewPatterns
-ViewR
-Visible
-V.null
-VRML
-V.toList
-W3CDTF
-Wait
-Walkable
-Wang
-Warning
-Warnings
-WARRANTY
-Was
-We
-WebTeX
-Whether
-While
-While1
-WhileEnd
-WhitePoint
-Width
-WidthParameter
-Widths
-Will
-Windows
-Windows.
-WINDOWS
-With
-With3
-WithAttrs
-WithClass
-WithIds
-WithIndex
-WithLang
-WithM
-WithName
-WithNewline
-WithOrderedListMarker
-Without
-WITHOUT
-WithoutClasses
-WithParameters
-WithSpace
-WithWarnings
-Word
-Word16
-Word16be
-Word16le
-Word32
-Word32be
-Word32le
-Word64
-Word64be
-Word64le
-Word8
-WordChar
-Words
-Work
-Wrap
-Wrapper
-Wrapping
-Wraps
-WrapText
-WriteMode
-Writer
-WriterEnvironment
-WriterOptions
-WriterReader
-Writers
-WriterState
-WrongElem
-WS
-X.Attr
-XC
-XC.current
-XC.Cursor
-XC.fromContent
-XC.modifyContent
-XC.nextDF
-XC.root
-XC.toTree
-XhtmlFormatter
-XL
-Xml
-XML
-XML.
-XML.Attr
-Xml.defaultConfigPP
-XMLDoc
-XMLDocE
-Xml.elAttribs
-XML.Elem
-XML.Element
-Xml.ppcElement
-Xml.qPrefix
-XMLSchema
-XML_Schema_Fictionbook_2.1
-Xml.useShortEmptyTags
-XResolution
-XSLT
-XXX
-XZ
-Yaml
-YAML
-Yaml.Array
-Yaml.Bool
-Yaml.decodeEither
-YamlException
-YamlLines
-YamlMark
-Yaml.Null
-Yaml.Number
-Yaml.Object
-YamlParseException
-Yaml.String
-Yaml.Value
-Yaml.YamlMark
-YCbCrCoefficients
-YCbCrPositioning
-You
-YResolution
-YYYY
-Z0
-Zone
-ZonedTime
```

Comment:
Are these really useful to discover? Would you immediately think of using A97C to represent your data? Datatypes are abstractions to represent data and are usually very specific to the situation. Ususally, the constructors are hidden from you. The most you can do is to use methods from that module to get the type and use it in your type signatures. The list does not include those methods, even. Hence, the benefits of using this list for hangman is greatly minimised.

Compare this with my list, the Haskell Prelude, which defines most of the most commonly used methods in Haskell. For example, zip. Zip is a common idiom in Haskell to pair elements from two lists together. Usually, people use list comprehension or map to achieve the result, and those methods are less concise. Hence, by using this list, people will find out about zip, a common idiom, to help them reduce duplication, reinventing the wheel and work, which ultimately benefits them.

This is why this change is better/necessary to remain true to the objectives of the game.